core/renderers/zelos/drawer.js

```diff
@@ -45,3 +45,63 @@ Blockly.zelos.Drawer = function(block, info) {
 };
 goog.inherits(Blockly.zelos.Drawer, Blockly.blockRendering.Drawer);
 
+
+/**
+ * Add steps for the top corner of the block, taking into account
+ * details such as hats and rounded corners.
+ * @protected
+ */
+Blockly.zelos.Drawer.prototype.drawTop_ = function() {
+  var topRow = this.info_.topRow;
+  var elements = topRow.elements;
+
+  this.positionPreviousConnection_();
+  this.outlinePath_ +=
+      Blockly.utils.svgPaths.moveBy(topRow.xPos, this.info_.startY);
+  for (var i = 0, elem; (elem = elements[i]); i++) {
+    if (elem.type == 'round corner') {
```

Comment:
I prefer 'left round corner' and 'right round corner' to 'round corner' and 'right round corner'.

If you've got ideas about better ways of handling types (that doesn't involve instanceof) lmk.

Reply:
I would prefer that as well. I wanted to leave the types as is ``round corner`` instead of adding the left, as there's a few places that check the type directly rather than using the base method ``isRoundCorner``.

In terms of dealing with types, here are a few options:
- do what we're doing now, con with that is that the check method is on the measurable, ie: to add additional types you have to do them in base measurable, otherwise ``elem.isSpecialType()`` might not exist on all elements.
- factory type checker, something like: ``Blockly.isSpecialType(elem)`` that gets passed in an element and checks the type. This can be extended independently in each renderer to types that only matter there.
- Use a enum flag and bitwise operations with types. This is the option I prefer, but it might be an overkill. We could have a central place to register types, which basically internally appends new types to a enum flag. Then you can do things like: 
```
if (elem.type & Blockly.Measurable.Types['left row']) {
}
```
The nice thing about enum flags is you can have union types, so you can do bitwise checks to find if an element is a corner (not really caring about the specific type of corner it is), but then again you could always just do indexOf('corner') haha.

A combination of 2 and 3 would be nice too.

Reply:
I created a tracking bug: #2946

Reply:
Thanks